### ISSUE TITLE
Packaging help for versions >= 0.8

### ISSUE THREAD

**Issue by adigitoleo** [indygreg/PyOxidizer#311]
I'm writing an updated Arch Linux package for this project since [the existing one](https://aur.archlinux.org/packages/pyoxidizer) is way back at 0.3 (and has the wrong license). I'll be pulling the sources directly from the main branch since it's supposed to be stable (so it will be a "VCS package" in Arch terminology). It's my first PKGBUILD and first time using cargo. I have a couple of problems:

- I need to use `cargo build` and then install using pacman in a separate step. I read [the docs](https://pyoxidizer.readthedocs.io/en/latest/rust_cargo_source_checkouts.html#rust-cargo-source-checkouts) and from what I understand, if I just want to create executables from python I just need the base `pyoxidizer` crate. So I grabbed the last commit from `main` and ran `cargo build --release --locked --bin pyoxidizer`. But then `cargo test --release --locked --bin pyoxidizer` fails at `linux_extensions_build_with_library`. I also tried the tests again adding `-p pyoxidizer` but it still fails.

- I'm parsing `git describe --long --tags` to build the version ID, which interestingly gives me v0.8.xxx instead of v0.9.xxx (#295 related?). Also, when I quickly checked the tag list I saw `oxidized_importer-0.1`. That kind of nonstandard tag would break my PKGBUILD if it cropped up again. Can we stick to [semantic versioning](https://semver.org/) please ;)

Any other tips/recommendations are welcome.

**Comment by indygreg** [comment]
>     * I need to use `cargo build` and then install using pacman in a separate step. I read [the docs](https://pyoxidizer.readthedocs.io/en/latest/rust_cargo_source_checkouts.html#rust-cargo-source-checkouts) and from what I understand, if I just want to create executables from python I just need the base `pyoxidizer` crate. So I grabbed the last commit from `main` and ran `cargo build --release --locked --bin pyoxidizer`. But then `cargo test --release --locked --bin pyoxidizer` fails at `linux_extensions_build_with_library`. I also tried the tests again adding `-p pyoxidizer` but it still fails.

This sounds like a test is failing? If so, please file an issue with more details so we can track down why that test is failing.

>     * I'm parsing `git describe --long --tags` to build the version ID, which interestingly gives me v0.8.xxx instead of v0.9.xxx (#295 related?). Also, when I quickly checked the tag list I saw `oxidized_importer-0.1`. That kind of nonstandard tag would break my PKGBUILD if it cropped up again. Can we stick to [semantic versioning](https://semver.org/) please ;)

Yeah, `git describe` won't work on this repo unless you use the `--match` or `--exclude` arguments. I'd recommend you parse the output of `pyoxidizer -V` or `pyoxidizer --version` as an alternative. Or parse the `version` key from `pyoxidizer/Cargo.toml` (this is essentially how PyOxidizer crafts its own version string).

**Comment by adigitoleo** [comment]
OK I'm going to have to parse that .toml file then, since I need the version before starting the build. My unix-fu is not quite there yet, I tried:

```sh
# Construct a unique pkgver for each commit on the main branch, in the format:
# RELEASE.rREVISION

local toml=$(awk '/\[/{section=$0; next} $1{print section $0}' Cargo.toml)
for line in $toml; do
    case $line in
        \[package\]version=* )
            printf "%s.r%s.%s" \
                "$(echo $line | sed 's/^\[package\]version=//;s/"//;s/-/_')" \
                "$(git rev-list --count HEAD)" \
                "$(git rev-parse --short HEAD)"
            ;;
    esac
done
```

but it's not working, I'll need to fiddle with it some more when I get time. Thanks for the pointers.

**Comment by adigitoleo** [comment]
OK I asked at the Arch forums and fixed the toml parsing:

```sh
    printf "%s.r%s.%s" \
        $(sed -n '/version/{s/.*"\([0-9\.]*\).*"/\1/p;q}' Cargo.toml) \
        "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
```

Now it't time to open another issue for the failing test.
